pkg/lib/docker/docker.go

```diff
@@ -47,13 +48,22 @@ var NoAuth string
 
 var _cachedClient *Client
 
+func init() {
+	NoAuth, _ = EncodeAuthConfig(dockertypes.AuthConfig{})
+}
+
 type Client struct {
 	*dockerclient.Client
 	Info dockertypes.Info
 }
 
-func init() {
-	NoAuth, _ = EncodeAuthConfig(dockertypes.AuthConfig{})
+type K8sAuthSecret struct {
```

Comment:
Could this struct definition be inlined in getDockerAuthStrFromK8s? It looks it is only used for unmarshalling right?